reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md

```diff
@@ -0,0 +1,682 @@
+qualified command name instead.
+
+```csharp
+// Trigger on success
+if (target == FeedbackTrigger.Success){
+    // Getting the commands from the AST and only finding those that are Commands
+    var astCmds = ast.FindAll((cAst) => cAst is CommandAst, true);
+
+    // Check to see if there are any command returned
+    if(astCmds is null || astCmds.Count() == 0){
+        return null;
+    }
+
+    // Inspect each of the commands
+    foreach(var command in astCmds){
+
+        // Get the command name
+        var aliasedCmd = ((CommandAst) command).GetCommandName();
+
+        // Check if its an alias or not, if so then add it to the list of actions
+        if(TryGetAlias(aliasedCmd, out string commandString)){
+            actions.Add(aliasedCmd + " --> " + commandString);
+
+        }
+
+    }
+
+    // If no aliases were found return null
+    if(actions.Count == 0){
+        return null;
+    }
+
+    // If aliases are found, set the header to a description and return a new FeedbackItem.
+    header = "You have used an aliased command:";
+    // Copy actions to _candidates for the predictor
+    _candidates = actions;
+
+    return new FeedbackItem(header, actions);
+}
+```
+
+### Implement the TryGetAlias() method
+
+The `TryGetAlias()` method is a private helper function that returns a boolean value to indicate
+whether the command is an alias. In the class constructor, we created a PowerShell instance that we
+can use to run PowerShell commands. The `TryGetAlias()` method uses this PowerShell instance to
+invoke the `GetCommand` method to determine if the command is an alias. The `AliasInfo` object
+returned by `GetCommand` contains full name of the aliased command. The parameters of
+`TryGetAlias()` are: `command`, which is the command to be checked, and `targetCommand`, which is
+the output variable to contain the full name of the aliased command.
+
+```csharp
+private bool TryGetAlias(string command, out string targetCommand){
+    // Create PowerShell runspace as a session state proxy to run GetCommand and check
+    // if its an alias
+    AliasInfo? pwshAliasInfo =
+        _powershell.Runspace.SessionStateProxy.InvokeCommand.GetCommand(command, CommandTypes.Alias) as AliasInfo;
+
+    // if its null then it is not an aliased command so just return false
+    if(pwshAliasInfo is null){
+        targetCommand = String.Empty;
+        return false;
+    }
+
+    // Set targetCommand to referenced command name
+    targetCommand = pwshAliasInfo.ReferencedCommand.Name;
+    return true;
+}
+```
+
+### Create suggestions for a Failure trigger
+
+When a command execution fails, we want to suggest that the user `Get-Help` to get more information
+about how to use the command.
+
+```csharp
+// Trigger on error
+if (target == FeedbackTrigger.Error){
+    // Gets the command that caused the error.
+    var erroredCommand = context.LastError?.InvocationInfo.MyCommand;
+    if (erroredCommand is null){
+        return null;
+    }
```

Comment:
You have `var commandLine = context.CommandLine;` above, which is the command line that caused the error.

Reply:
Not sure what you mean?

Reply:
I want to show off the last error property of `context`, I figured this would be easier than trying to grab the cmd in the string of the commandline

Reply:
Ah, right. We are good here.